route_following_plugin/include/route_following_plugin.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <vector>
+#include <cav_msgs/Plugin.h>
+#include <carma_utils/CARMAUtils.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+#include <carma_wm/WMListener.h>
+#include <carma_wm/WorldModel.h>
+#include <cav_srvs/PlanManeuvers.h>
+
+namespace route_following_plugin
+{
+
+    class RouteFollowingPlugin
+    {
+
+    public:
+
+        // constant speed limit for test
+        // TODO Once world_model and vector map is ready, it should be removed
```

Comment:
We might want to think about whether this data will be available by the time this integration branch gets merged into develop. I think this world model and vector map data is a requirement for the tactical plugin to work, and thus we'll need that to function before we can merge this. Since the interface is already there, is it possible to start using the speed limits from that in this strategic plugin?

Reply:
On one hand, there is no interface in the world_model to expose traffic rule. I believe it is still in progress. One the other hand, I am not sure if Samir finished his task on adding speed limit into vector map. Those two things are what I am waiting on. I agree it should not be merged into develop before we have those. Right now it targets to the routing integration branch.

Reply:
I'm not sure if we have a separate story for integrating that, or how the best way to track that is then. I just want to make sure we don't lose track of that somewhere as we'll really need to use that functionality once that data is available. Do you think that fits within the scope of another story, I don't want to hold this one open for a single issue if that's avoidable.

Reply:
Does it make sense to do it under that wrap-all integration test? We probably are ready at that time.